src/Surfnet/StepupMiddleware/ApiBundle/Configuration/Projector/InstitutionAuthorizationProjector.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/**
+ * Copyright 2018 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupMiddleware\ApiBundle\Configuration\Projector;
+
+use Broadway\ReadModel\Projector;
+use Surfnet\Stepup\Configuration\Event\InstitutionConfigurationRemovedEvent;
+use Surfnet\Stepup\Configuration\Event\NewInstitutionConfigurationCreatedEvent;
+use Surfnet\Stepup\Configuration\Event\SelectRaaOptionChangedEvent;
+use Surfnet\Stepup\Configuration\Event\UseRaaOptionChangedEvent;
+use Surfnet\Stepup\Configuration\Event\UseRaOptionChangedEvent;
+use Surfnet\Stepup\Configuration\Value\Institution;
+use Surfnet\Stepup\Identity\Collection\InstitutionCollection;
+use Surfnet\Stepup\Identity\Event\InstitutionsAddedToWhitelistEvent;
+use Surfnet\Stepup\Identity\Event\InstitutionsRemovedFromWhitelistEvent;
+use Surfnet\Stepup\Identity\Event\WhitelistCreatedEvent;
+use Surfnet\Stepup\Identity\Event\WhitelistReplacedEvent;
+use Surfnet\StepupMiddleware\ApiBundle\Configuration\Repository\InstitutionAuthorizationRepository;
+use Surfnet\StepupMiddleware\ApiBundle\Configuration\Repository\InstitutionConfigurationOptionsRepository;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+final class InstitutionAuthorizationProjector extends Projector
+{
+    /**
+     * @var InstitutionAuthorizationRepository
+     */
+    private $institutionAuthorizationRepository;
+    /**
+     * @var InstitutionConfigurationOptionsRepository
+     */
+    private $institutionConfigurationOptionsRepository;
+
+    public function __construct(
+        InstitutionAuthorizationRepository $institutionAuthorizationRepository,
+        InstitutionConfigurationOptionsRepository $institutionConfigurationOptionsRepository
+    ) {
+        $this->institutionAuthorizationRepository = $institutionAuthorizationRepository;
+        $this->institutionConfigurationOptionsRepository = $institutionConfigurationOptionsRepository;
+    }
+
+    public function applyNewInstitutionConfigurationCreatedEvent(NewInstitutionConfigurationCreatedEvent $event)
```

Comment:
Now that I'm able to do some functional testing, I strongly suspect that this event listener is not required in this projection.

When a new whitelist entry is created the `InstitutionsAddedToWhitelistEvent` is triggered and defaults are created in this projection. But the `NewInstitutionConfigurationCreatedEvent` is also triggered (as the other default configuration options are also created (The where how and why remains a mystery which you will unravel)).

Reply:
I think we should only listen to` InstitutionConfiguration` -events and not to the `Whitelist` -events. In pivotal I'm trying to find out the best solution.